kasa/iotprotocol.py

```diff
@@ -0,0 +1,106 @@
+"""Module for the IOT legacy IOT KASA protocol."""
+import asyncio
+import logging
+from typing import Dict, Optional, Union
+
+import httpx
+
+from .credentials import Credentials
+from .exceptions import AuthenticationException, SmartDeviceException
+from .json import dumps as json_dumps
+from .klaptransport import TPLinkKlapTransport
+from .protocol import TPLinkProtocol, TPLinkTransport
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TPLinkIotProtocol(TPLinkProtocol):
+    """Class for the legacy TPLink IOT KASA Protocol."""
+
+    DEFAULT_PORT = 80
+
+    def __init__(
+        self,
+        host: str,
+        *,
+        transport: Optional[TPLinkTransport] = None,
+        credentials: Optional[Credentials] = None,
+        timeout: Optional[int] = None,
+    ) -> None:
+        super().__init__(host=host, port=self.DEFAULT_PORT)
+
+        self.credentials: Credentials = (
+            credentials
+            if credentials and credentials.username and credentials.password
+            else Credentials(username="", password="")
+        )
+        self.transport: TPLinkTransport = (
+            transport
+            if transport
+            else TPLinkKlapTransport(
+                host, credentials=self.credentials, timeout=timeout
+            )
+        )
```

Comment:
See my other comment wrt. using a simple `or` for these.

Reply:
Done